### ISSUE TITLE
newest pixi-tilemap fails with renderer.plugins.tilemap undefined

### ISSUE THREAD

**Issue by blurymind** [pixijs/tilemap#124]
I am currently in the process of upgrading pixi-tilemap on gdevelop and after updating to the newest version of pixi-tilemap, for some reason it no longer gets added to plugins.

https://github.com/blurymind/GDevelop/tree/ldtk-p6
https://github.com/blurymind/GDevelop/blob/ldtk-p6/Extensions/TileMap/pixi-tilemap-helper.js#L586

You can test this by changing between the old and the new version of the module.The old one works fine, the new one fails here:

https://github.com/pixijs/tilemap/blob/master/dist/pixi-tilemap.umd.js#L468
and everywhere else, because
renderer.plugins.tilemap is undefined

**Comment by eXponenta** [comment]
Tilemap MUST BE registered before renderer initialisation initialisation and requred global PIXI context.
For other you should register render plygin manually like:
https://github.com/pixijs/tilemap/blob/master/src/TileRenderer.ts#L201

**Comment by blurymind** [comment]
how does one do that on an existing Pixi instance? I dont think I can add it when pixi is initiated

**Comment by ivanpopelyshev** [comment]
>  I dont think I can add it when pixi is initiated

so, pixi-tilemap is included after renderer is created?

```js
renderer.plugins.tilemap = new TileRenderer(renderer);
```

something like that

**Comment by ivanpopelyshev** [comment]
I honestly didnt think about this case

**Comment by blurymind** [comment]
Yes unfortunately because of how the extensions in Gdevelop work, the external library is part of the extension itself

**Comment by blurymind** [comment]
`renderer.plugins.tilemap = new TileRenderer(renderer);`

I can access PIXI's global object, but have no idea how to get to the renderer.
PIXI.Renderer? thats a constructor, how do you get the current renderer?

**Comment by ivanpopelyshev** [comment]
honestly, i dont know where Renderer is in GDevelop.

**Comment by ivanpopelyshev** [comment]
You can hack the method `render()` and if there's no plugin - create one.

**Comment by blurymind** [comment]
I think it would be better if I get the renderer from gdevelop itself

**Comment by blurymind** [comment]
@ivanpopelyshev I managed to redister it as a plugin, however now there is another problem
![image](https://user-images.githubusercontent.com/6495061/126538202-f1d10026-bf4b-472a-a56a-0eac9e179b7e.png)

is this because our version of pixi on GD doesnt have utils? I am confused

**Comment by blurymind** [comment]
@ivanpopelyshev I might have discovered a bug.
utils appears to be in utils.utils for that line

**Comment by blurymind** [comment]
here is my wip pr btw https://github.com/4ian/GDevelop/pull/2828

**Comment by ivanpopelyshev** [comment]
Ok, so, apparantly in UMD build referece to utils might be wrong, i have to check it!

Btw, since you are here , tell me, is GDevelop using `PIXI.Graphics`?

**Comment by blurymind** [comment]
could be a better question for @4ian , but I think YES. I can find many references to it

**Comment by blurymind** [comment]
@ivanpopelyshev @4ian interestingly its 
`utils.utils.createIndicesForQuads `when I try to use the library from the IDE and its
`utils.createIndicesForQuads `when I do it from the runtime

**Comment by eXponenta** [comment]
> @ivanpopelyshev I might have discovered a bug.
> utils appears to be in utils.utils for that line. Changing it to
> 
> ```
>     	    this.indexBuffer.update(utils.utils.createIndicesForQuads(size,
>     	        settings.use32bitIndex ? new Uint32Array(size * 6) : undefined));
> ```
> 
> seems to fix it for me

Very strange, because bundler link valid `utils` to UMD.
Look to last argument that passed as utils. it is `this.PIXI.utils`, where `this` is context global object  (self for worker, global for node, window for browser)

https://github.com/pixijs/tilemap/blob/master/dist/pixi-tilemap.umd.js#L17
